google-oslogin-trustedca.timer

```diff
@@ -0,0 +1,9 @@
+[Unit]
+Description=OS Login Trusted CA SSH keys sync timer
+
+[Timer]
+OnBootSec=5
+OnUnitActiveSec=5m
```

Comment:
Wow, so many questions for that one thing...

> where's 5m coming from?

There was a discussion in the design document (where you're tagged as an approver by the way). So, I'm not repeating the same, please read my comment from `2:55 PM Apr 26`. If there are still some questions left, I'll definitely reply, but first, please read that doc.

> What happens if the feature is enabled 1 second after this timer runs?

There is no such a thing as "enabling" that particular feature. It's gonna be either enabled with the OS Login it self or  disabled. If OS Login disabled, Guest Agent removes `all` options from sshd_config, including this new feature. But, if OS Login is enabled, then all OS Login related features are enabled at once.

> Will the user be locked outside for 5m?

Of course not. Consulting with systemd.timer(5) it turns out that `OnBootSec=5` ensures that after 5 sec since boot this timer will be triggered and all CA SSH keys will be fetched and placed in the appropriate file. There is no 5 min waiting for the first fetch. It's done asynchronously to the feature activation, and one doesn't depend on another. Guest Agent separately activates OS Login if case it's enabled in the metadata and configured SSHD to watch the files with trusted CA keys - even if the file doesn't exist at that moment! That's important. So, as soon as Guest Agent notices that OS Login is enabled, SSHD gets configured to support Trusted CA keys. And once this file occurs in the defined place, it will be used with no pause.

The tool from this particular review, does the synchronization of trusted CA keys, and it does this again asynchronously to the feature activation. Once the OS is booted, in 5 sec or as close as possible to that time, the timer will be triggered and the tool update CA keys.

There is no waiting, no matter what is the period defined in `OnUnitActiveSec`

> should the user force restarting the vm after enabling it?

What? Restart of a VM? It seems you're missing some crucial concept here. 
Guest Agent keeps an eye on the metadata flag of OS Login and enables/disables OS Login on a VM dynamically.

> the same questions apply to guest-agent changes to bootstrap the config as well

Of course not. SSHD configuration is updated on the fly by Guest Agent. That's the whole point of having Guest Agent on VMs.